src/imu_full_measured_data_ros_publisher.cpp

```diff
@@ -0,0 +1,96 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "imu_ros2/imu_full_measured_data_ros_publisher.h"
+#include <thread>
+#include <chrono>
+
+ImuFullMeasuredDataRosPublisher::ImuFullMeasuredDataRosPublisher(std::shared_ptr<rclcpp::Node>& node)
+{
+    init(node);
+}
+
+ImuFullMeasuredDataRosPublisher::~ImuFullMeasuredDataRosPublisher()
+{
+    delete m_dataProvider;
+}
+
+void ImuFullMeasuredDataRosPublisher::init(std::shared_ptr<rclcpp::Node> &node)
+{
+    m_node = node;
+    m_publisher = node->create_publisher<imu_ros2::msg::ImuFullMeasuredData>("imufullmeasureddata", 10);
+}
+
+void ImuFullMeasuredDataRosPublisher::setMessageProvider(ImuFullMeasuredDataProviderInterface *dataProvider)
+{
+    m_dataProvider = dataProvider;
+}
+
+void ImuFullMeasuredDataRosPublisher::run()
+{
+    std::thread::id this_id = std::this_thread::get_id();
+    std::cout << "thread " << this_id << " started...\n";
+    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "startThread: '%d'", this_id);
+
+    typedef std::chrono::system_clock::time_point timenow;
+    timenow started;
+    timenow done;
+    bool success = false;
+
+    rclcpp::WallRate loopRate(0.1);
+
+    while (rclcpp::ok()) {
+
+        int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
+
+        switch(operation_mode) {
+        case DEVICE_CONTINUOUS_SAMPLING_MODE:
+            started = std::chrono::high_resolution_clock::now();
+            success = m_dataProvider->getData(m_message);
+            done = std::chrono::high_resolution_clock::now();
+
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu acceleration data: '%f' '%f' '%f'",
+                        m_message.accel.x, m_message.accel.y, m_message.accel.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu gyroscope data: '%f' '%f' '%f'",
+                        m_message.gyro.x, m_message.gyro.y, m_message.gyro.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu delta velocity data: '%f' '%f' '%f'",
+                        m_message.delta_vel.x, m_message.delta_vel.y, m_message.delta_vel.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu delta angle data: '%f' '%f' '%f'",
+                        m_message.delta_angle.x, m_message.delta_angle.y, m_message.delta_angle.z);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu temperature data: '%f' ",
+                        m_message.temp);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_full_measured_data"), "Publishing imu time %d us ",
+                        std::chrono::duration_cast<std::chrono::microseconds>(done-started).count());
```

Comment:
I don't think this data should be logged always. if success is false we should log that success is false and read data is not valid. is success is true we can log the values